src/components/Composer.vue

```diff
@@ -372,6 +393,37 @@ export default {
 		onAddCloudAttachment() {
 			this.bus.$emit('onAddCloudAttachment')
 		},
+		onSaveAndClose() {
+			this.state = STATES.SAVING
+			this.saveDraft(this.getMessageData)
+			wait(1500) // Illusion of things going on behind the scenes
+				.then(() => (this.state = STATES.SAVED))
+				.then(() => wait(1000)) // Time to register the "success"-message
+				.then(() => this.$router.push({name: 'home'})) // TODO: Navigate to better place
+		},
+		onDelete() {
+			this.state = STATES.DISCARDING
+			this.draftsPromise
+				.then(this.getMessageData)
+				.then(message => {
+					// TODO: Somehow delete the draft
+					// We've got `message`, but we need an `envelope`. I'm not 100%
+					// on the distinction yet, but `envelope` seems to have some
+					// necessary parameters which we don't have access to here.
+					// const envelope = magic(message)
+
+					// I'm not sure if this is needed. It only seems to affect the
+					// Mailbox.vue, which I doooon't think is relevant from within
+					// a draft? Maaybe?
+					// this.$emit('delete', {envelope})
```

Comment:
It's needed when you *resume a draft*, as in open an existing draft in your *Drafts* mailbox. Then you'll see the message composer and work on something that is actually displayed in the second column. In that case it makes sense to emit the `delete` event, so that the message is removed from the list.